src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/repository/FlattenedSecurityServerClientRepository.java

```diff
@@ -0,0 +1,317 @@
+                        predicates.add(subsystemPredicate(root, builder));
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Invalid client type " + params.getClientType());
+                }
+            }
+            if (predicates.isEmpty()) {
+                predicates.add(idIsNotNull(root, builder));
+            }
+            return builder.and(predicates.toArray(new Predicate[predicates.size()]));
+        };
+    }
+    
+    static Specification<FlattenedSecurityServerClient> instance(String s) {
+        return (root, query, builder) -> {
+            return instancePredicate(root, builder, s);
+        };
+    }
+
+    static Specification<FlattenedSecurityServerClient> memberClass(String s) {
+        return (root, query, builder) -> {
+            return memberClassPredicate(root, builder, s);
+        };
+    }
+
+    static Specification<FlattenedSecurityServerClient> memberCode(String s) {
+        return (root, query, builder) -> {
+            return memberCodePredicate(root, builder, s);
+        };
+    }
+
+    static Specification<FlattenedSecurityServerClient> memberName(String s) {
+        return (root, query, builder) -> {
+            return memberNamePredicate(root, builder, s);
+        };
+    }
+
+    static Specification<FlattenedSecurityServerClient> subsystemCode(String s) {
+        return (root, query, builder) -> {
+            return subsystemCodePredicate(root, builder, s);
+        };
+    }
+
+    static Specification<FlattenedSecurityServerClient> member() {
+        return (root, query, builder) -> {
+            return memberPredicate(root, builder);
+        };
+    }
+
+    static Specification<FlattenedSecurityServerClient> subsystem() {
+        return (root, query, builder) -> {
+            return subsystemPredicate(root, builder);
+        };
+    }
+
+    static Specification<FlattenedSecurityServerClient> securityServerId(int id) {
+        return (root, query, builder) -> {
+            return clientOfSecurityServerPredicate(root, builder, id);
+        };
+    }
+
+    static Specification<FlattenedSecurityServerClient> multifieldSearch(String q) {
+        return (root, query, builder) -> {
+            return multifieldTextSearchPredicate(root, builder, q);
+        };
+    }
+
+    private static Predicate memberPredicate(Root root, CriteriaBuilder builder) {
+        return builder.equal(root.get(FlattenedSecurityServerClient_.TYPE), "XRoadMember");
+    }
+    private static Predicate subsystemPredicate(Root root, CriteriaBuilder builder) {
+        return builder.equal(root.get(FlattenedSecurityServerClient_.TYPE), "Subsystem");
+    }
+    private static Predicate memberNamePredicate(Root root, CriteriaBuilder builder, String s) {
+        return builder.like(
+                builder.lower(root.get(FlattenedSecurityServerClient_.MEMBER_NAME)),
+                builder.lower(builder.literal("%" + s + "%"))
```

Comment:
Need to escape special characters like %, see https://github.com/nordic-institute/X-Road/pull/1207

Reply:
Addressed in https://github.com/nordic-institute/X-Road/commit/6ad7a88a6c78b661279cd1158f56ce0ba8cdd4f8